### PR TITLE
Introduce Lotus::Controller::Error

### PR DESCRIPTION
**Pull request by karimmtarek**
- Added Error class
- Replaced ::StandardError with Lotus::Controller::Error
- Added a test

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -435,7 +435,7 @@ action.call({id: 'unknown'}) # => raises RecordNotFound
 #### Inherited Exceptions
 
 ```ruby
-class MyCustomException < StandardError
```
Comment (jodosha) [inline README.md]: @karimmtarek This is unrelated to the scope of your change, please revert it. :smile:

File: README.md [hunk 2]
```diff
@@ -458,7 +458,7 @@ module Articles
   class Show
     include Lotus::Action
 
-    handle_exception StandardError => :handle_standard_error
+    handle_exception Lotus::Controller::Error => :handle_controller_error
```
Comment (jodosha) [inline README.md]: @karimmtarek This is unrelated to the scope of your change, please revert it. :smile:

File: test/fixtures.rb [hunk 3]
```diff
@@ -107,11 +107,11 @@ def call(params)
   end
 end
 
-class MyCustomError < StandardError ; end
```
Comment (jodosha) [inline test/fixtures.rb]: @karimmtarek All the changes in this file are unrelated to the scope of your change, please revert all of them. :smile:

File: test/integration/rack_errors_test.rb [hunk 4]
```diff
@@ -9,10 +9,10 @@
   get '/framework_managed',   to: 'errors#framework_managed'
 end
 
-HandledException          = Class.new(StandardError)
```
Comment (jodosha) [inline test/integration/rack_errors_test.rb]: @karimmtarek All the changes in this file are unrelated to the scope of your change, please revert all of them. :smile:

File: README.md (hunk 1 above)
Comment (karimmtarek) [inline README.md]: @jodosha Sure I'll revert to the previous readme file :)

**Comment by jodosha** [comment]
@karimmtarek Hey thanks for this PR. :smile: 

You replaced all the occurrences of `StandardError` in the project, but this is unrelated to the scope of your change. With that tests and fixtures we're verifying the behavior of the framework when an exception that comes from the application.

This commit is to revert entirely. :disappointed: 

The only change that you should made is for `Lotus::Controller::UnknownFormatError` that should inherit from `Lotus::Controller::Error` instead of `StandardError`.

Can you please fix it? Thanks!

File: test/fixtures.rb (hunk 3 above)
Comment (karimmtarek) [inline test/fixtures.rb]: @jodosha reverting the change for `ErrorCallWithStringMethodNameAsHandlerAction` will result in a failing test in `action_test.rb`, this one 
```
it 'handles exception with specified method (string)' do
  response = ErrorCallWithStringMethodNameAsHandlerAction.new.call({})

  response[0].must_equal 502
  response[2].must_equal ['Lotus::Controller::Error']
end
```

unless I revert the change there as well, please advice, thanks.

Comment (jodosha) [inline test/fixtures.rb]: @karimmtarek What's the failure?

**Comment by karimmtarek** [comment]
@jodosha Sorry for missing things up :disappointed: 
I'll revert, and see how it goes.

**Comment by karimmtarek** [comment]
@jodosha what about now?

File: test/error_test.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+describe Lotus::Controller::Error do
+  it 'inherits from ::StandardError' do
+    Lotus::Controller::Error.superclass.must_equal StandardError
```
Comment (jodosha) [inline test/error_test.rb]: @karimmtarek It lacks of test case for `Lotus::Controller::UnknownFormatError` :wink:

Comment (karimmtarek) [inline test/error_test.rb]: @jodosha Done :+1:

**Comment by jodosha** [comment]
@karimmtarek It looks good now. Thank you! :+1:
